app/services/PayeAtsService.scala

```diff
@@ -17,83 +17,82 @@
 package services
 
 import com.google.inject.Inject
-import com.typesafe.scalalogging.LazyLogging
 import connectors.MiddleConnector
 import controllers.auth.{AuthenticatedRequest, PayeAuthenticatedRequest}
-import models.PayeAtsData
-import play.api.{Logger, Logging}
-import play.api.http.Status.{BAD_REQUEST, INTERNAL_SERVER_ERROR, NOT_FOUND, OK}
-import play.api.libs.json.Reads
+import models._
+import play.api.Logging
+import play.api.http.Status.{BAD_REQUEST, NOT_FOUND}
 import uk.gov.hmrc.domain.Nino
-import uk.gov.hmrc.http.{BadRequestException, HeaderCarrier, HttpResponse, NotFoundException}
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.audit.http.connector.AuditResult
 import utils.AuditTypes
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
 
 class PayeAtsService @Inject()(middleConnector: MiddleConnector, auditService: AuditService)(
   implicit ec: ExecutionContext)
     extends Logging {
 
   def getPayeATSData(nino: Nino, taxYear: Int)(
     implicit hc: HeaderCarrier,
-    request: PayeAuthenticatedRequest[_]): Future[Either[HttpResponse, PayeAtsData]] =
-    middleConnector.connectToPayeATS(nino, taxYear) map { response =>
-      handleConnectorResponse[PayeAtsData](response, nino, taxYear)
-    } recover {
-      case e: BadRequestException => Left(HttpResponse(BAD_REQUEST, e.getMessage))
-      case e: NotFoundException   => Left(HttpResponse(NOT_FOUND, e.getMessage))
-      case e: Exception =>
-        logger.error(s"Exception in PayeAtsService: $e", e)
-        Left(HttpResponse(INTERNAL_SERVER_ERROR, e.getMessage))
+    request: PayeAuthenticatedRequest[_]): Future[Either[AtsResponse, PayeAtsData]] =
+    for {
+      response <- middleConnector.connectToPayeATS(nino, taxYear)
+    } yield {
+      response match {
```

Comment:
```suggestion
    middleConnector.connectToPayeATS(nino, taxYear).map {
```
I am wondering if this would be easier to read? No need for the match we can just pattern match directly in the map.